set.mm

```diff
@@ -556554,6 +556571,269 @@ range of a range Cartesian product ( ~ dfcoss4 ), but neither of these
+       Mazsa, 22-Aug-2021.) $)
+    elrelscnveq4 $p |- ( R e. Rels ->
+                         ( `' R C_ R <-> A. x A. y ( x R y <-> y R x ) ) ) $=
+      ( crels wcel ccnv wss wceq cv wbr wb wal elrelscnveq elrelscnveq2 bitrd )
+      CDECFZCGPCHAIZBIZCJRQCJKBLALCMABCNO $.
+  $}
+
+  $( The converse of a set is an element of the class of relations.
+     (Contributed by Peter Mazsa, 18-Aug-2019.) $)
+  cnvelrels $p |- ( A e. V -> `' A e. Rels ) $=
+    ( wcel ccnv crels wrel relcnv cvv wb cnvexg elrelsrel syl mpbiri ) ABCZADZE
+    CZOFZAGNOHCPQIABJOHKLM $.
+
+  $( Cosets of sets are elements of the relations class.  Implies
+     ` |- ( R e. Rels -> ,~ R e. Rels ) ` .  (Contributed by Peter Mazsa,
+     25-Aug-2021.) $)
+  cosselrels $p |- ( A e. V -> ,~ A e. Rels ) $=
+    ( wcel ccoss cvv crels cossex wrel relcoss elrelsrel mpbiri syl ) ABCADZECZ
+    MFCZABGNOMHAIMEJKL $.
+
+  $( Cosets of converse sets are elements of the relations class.  (Contributed
+     by Peter Mazsa, 31-Aug-2021.) $)
+  cosscnvelrels $p |- ( A e. V -> ,~ `' A e. Rels ) $=
+    ( wcel ccnv crels ccoss cnvelrels cosselrels syl ) ABCADZECJFECABGJEHI $.
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+                Subset relations
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  ${
+    $d x y $.
+    $( Define the subsets class or the class of all subset relations.  Similar
+       to definitions of epsilon relation ( ~ df-eprel ) and identity relation
+       ( ~ df-id ) classes.  The subset relation and subclass relationship
+       ( ~ df-ss ) are the same, that is, ` ( A _S B <-> A C_ B ) ` when ` B `
+       is a set, cf. ~ brssr .  Subset relation class and Scott Fenton's subset
+       class ~ df-sset are the same: ` _S = SSet ` (compare ~ dfssr2 with
+       ~ df-sset , cf. comment of ~ df-xrn ), the only reason we do not use
+       ~ dfssr2 as the base definition of the subsets class is the way we
+       defined the epsilon relation and the identity relation classes.
+
+       We need ` _S ` to define the class of reflexive sets so that we can
+       define the class of converse reflexive sets with the help of the
+       converse of ` _S ` , cf. ~~? df-refrels , ~~? df-refrel , ~~?
+       elrefrelsrel vs. ~~? df-cnvrefrels , ~~? df-cnvrefrel , ~~?
+       elcnvrefrelsrel .
+
+       The subsets class ` _S ` has another place in set.mm as well: if we
+       define extensional relation based on the common property in ~ extid ,
+       ~ extep and ~ extssr , then extrelssr $p |- ExtRel _S is a theorem along
+       with extrelep $p |- ExtRel _E and extrelid $p |- ExtRel _I .
+
+       "df-ss" is an alternative tag for the class of subsets ` _S ` since the
+       definition of subclass relation ` A C_ B ` , ~ df-ss should be "df-sc":
+       please consider renaming it/them for consistency.  (Contributed by Peter
+       Mazsa, 25-Jul-2019.) $)
```

Comment:
Personally I would agree that our current `df-ss` actually denotes "**s**ub**c**lass" and not "**s**ubt**s**et", so `sc` would be a better fitting name.
However it's maybe not just about `df-ss`, we have a whole lot of theorems where `ss` appears in the name.

I believe this deserves a larger discussion. I'm not sure what is the best way (an email on the list, or maybe a GitHub issue, since it seems most of the community's discussions take place on GitHub recently)

Reply:
Not very important in my opinion, but for this topic, the discussion group would probably be a better place.

Reply:
> However it's maybe not just about `df-ss`, we have a whole lot of theorems where `ss` appears in the name.

As noted at http://us.metamath.org/mpeuni/conventions-label.html we have adopted "ss" as the thing we put in the name. I'd be fine with saying "subclass" more in comments but I'm perhaps a bit more reluctant to change that many names (or, probably worse, change a few and not get around to most of them).

Reply:
I prefer the name `df-ss` for subset for `C_`, despite the fact that it applies to classes. It is much more common to refer to this as the subset relation than subclass, especially since most of the time the arguments are in fact sets (and for pragmatic reasons we don't want to need to use different operations). The way set.mm is set up, lots of things are technically classes despite morally (and provably) being sets, like `1` or `RR`. This has to do with the fact that there are no "set expressions": classes are expressions but there are only set variables in set.mm.

Reply:
@digama0 ok that would mean that we should keep labels (big relief!!) and this can be omitted from the boring-list (https is still on it:) Do you think we should mention this subclass/ss resolution in the comment section of df-ss? After that I would modify the text of the comment of df-ssr in my Mathbox by referring to df-ss.

Reply:
I'll do it tonight, mostly by copy&paste of your comment above.

Reply:
I`m happy that df-ss is not changed. I'll review this PR after the comments are revised.

Reply:
Please review (and feel free to modify) the new "Modification" commit below, esp. the comment of df-ss (which is of course in the main set.mm).

Reply:
The df-ss comment there looks OK to me. I think we may want to edit it in the future (for example to make it a bit shorter and to refer to some of our existing pages about classes, sets, terms, and variables), but I wouldn't hold up this pull request on that. I'd merge it now and worry about tweaking it later.